File: neetcode/src/test/java/solutions/arraysandhashing/LongestConsecutiveSequenceTest.java
package solutions.arraysandhashing;

import static org.junit.Assert.assertTrue;
import org.junit.Test;
import static solutions.arraysandhashing.LongestConsecutiveSequence.*;

public class LongestConsecutiveSequenceTest{
  @Test
  public void example1(){
    // init
    int[] input = {100,4,200,1,3,2};
    int expected = 4;
    int actual = longestConsecutive(input);

    // test
    String msg = "Expected " + expected + ", returned " + actual;
    assertTrue(msg, expected == actual);
  }

  @Test
  public void example2(){
    // init
    int[] input = {0,3,7,2,5,8,4,6,0,1};
    int expected = 9;
    int actual = longestConsecutive(input);

    // test
    String msg = "Expected " + expected + ", returned " + actual;
    assertTrue(msg, expected == actual);
  }
}

File: neetcode/python/test/arraysandhashing/longest_consecutive_sequence_test.py
import unittest
from typing import List
from src.arraysandhashing.longest_consecutive_sequence import Solution

class LongestConsecutiveSequenceTest(unittest.TestCase):
    def test_example1(self):
        nums = [100,4,200,1,3,2]
        actual = Solution.longestConsecutive(nums)
        expected = 4
        self.assertEqual(actual, expected)

    def test_example2(self):
        nums = [0,3,7,2,5,8,4,6,0,1]
        actual = Solution.longestConsecutive(nums)
        expected = 9
        msg = "The input contains duplicates that should not be counted"
        self.assertEqual(actual, expected, msg)

    def test_example3(self):
        nums = [0,1,5,6,7,8,9]
        actual = Solution.longestConsecutive(nums)
        expected = 5
        msg = "There are two sequences; the smaller is 2, the larger is 5"
        self.assertEqual(actual, expected, msg)

